src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -28,11 +28,12 @@ namespace Microsoft.AspNetCore.Http;
 /// <summary>
 /// Creates <see cref="RequestDelegate"/> implementations from <see cref="Delegate"/> request handlers.
 /// </summary>
-[UnconditionalSuppressMessage("Trimmer", "IL2026", Justification = "RequestDelegateFactory.Create requires unreferenced code.")]
-[UnconditionalSuppressMessage("Trimmer", "IL2060", Justification = "RequestDelegateFactory.Create requires unreferenced code.")]
-[UnconditionalSuppressMessage("Trimmer", "IL2072", Justification = "RequestDelegateFactory.Create requires unreferenced code.")]
-[UnconditionalSuppressMessage("Trimmer", "IL2075", Justification = "RequestDelegateFactory.Create requires unreferenced code.")]
-[UnconditionalSuppressMessage("Trimmer", "IL2077", Justification = "RequestDelegateFactory.Create requires unreferenced code.")]
+[UnconditionalSuppressMessage("Trimmer", "IL2026", Justification = "RequestDelegateFactory.Create has RequiresUnreferencedCode.")]
+[UnconditionalSuppressMessage("Trimmer", "IL2060", Justification = "RequestDelegateFactory.Create has RequiresUnreferencedCode.")]
+[UnconditionalSuppressMessage("Trimmer", "IL2072", Justification = "RequestDelegateFactory.Create has RequiresUnreferencedCode.")]
+[UnconditionalSuppressMessage("Trimmer", "IL2075", Justification = "RequestDelegateFactory.Create has RequiresUnreferencedCode.")]
+[UnconditionalSuppressMessage("Trimmer", "IL2077", Justification = "RequestDelegateFactory.Create has RequiresUnreferencedCode.")]
+[UnconditionalSuppressMessage("Trimmer", "IL3050", Justification = "RequestDelegateFactory.Create has RequiresDynamicCode.")]
```

Comment:
Instead of suppressing all these warnings for the whole class, can we instead put `RequiresUnreferencedCode` and `RequiresDynamicCode` at the class level? That will turn off any warnings in the class, and mark all static methods as "RUC"/"RDC".

That way if we add a new static method, it won't be missed to add the correct attribute to it.